src/databricks/labs/ucx/recon/migration_recon.py

```diff
@@ -0,0 +1,76 @@
+import logging
+from collections.abc import Iterable
+from dataclasses import dataclass
+from databricks.labs.lsql.backends import SqlBackend
+from databricks.labs.ucx.framework.crawlers import CrawlerBase
+from databricks.labs.ucx.hive_metastore.migration_status import MigrationStatusRefresher
+from databricks.labs.ucx.recon.base import (
+    DataComparator,
+    SchemaComparator,
+    TableIdentifier,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class ReconResult:
+    schema_matches: bool
+    data_matches: bool
+    schema_comparison: str
+    data_comparison: str
+
+
+class MigrationRecon(CrawlerBase[ReconResult]):
+    def __init__(
+        self,
+        sbe: SqlBackend,
+        schema,
+        migration_status_refresher: MigrationStatusRefresher,
+        schema_comparator: SchemaComparator,
+        data_comparator: DataComparator,
+    ):
+        super().__init__(sbe, "hive_metastore", schema, "recon_result", ReconResult)
+        self._migration_status_refresher = migration_status_refresher
+        self._schema_comparator = schema_comparator
+        self._data_comparator = data_comparator
+
+    def snapshot(self) -> Iterable[ReconResult]:
+        return self._snapshot(self._try_fetch, self._crawl)
+
+    def _crawl(self) -> Iterable[ReconResult]:
+        self._migration_status_refresher.reset()
+        for migration_status in self._migration_status_refresher.snapshot():
+            source = TableIdentifier(
+                "hive_metastore",
+                migration_status.src_schema,
+                migration_status.src_table,
+            )
+            if not self._migration_status_refresher.is_migrated(
+                migration_status.src_schema, migration_status.src_table
+            ):
```

Comment:
it's a strange logic - first we `refresher.snapshot():` and then we `refresher.is_migrated(..)`. Why don't you get an index and iterate over that?..